example/src/example.erl

```diff
@@ -0,0 +1,64 @@
+-module(example).
+
+-export([start/0]).
+-export([start/2]).
+-export([stop/0]).
+-export([stop/1]).
+
+%% application
+%% @doc Starts the application
+start() ->
+  {ok, _Started} = application:ensure_all_started(example).
+
+%% @doc Stops the application
+stop() ->
+  application:stop(example).
+
+%% behaviour
+%% @private
+start(_StartType, _StartArgs) ->
+  {ok, Pid} = example_sup:start_link(),
+  {ok, _Pid} = start_listeners(),
+  {ok, Pid}.
+
+%% @private
+stop(_State) ->
+  ok = cowboy:stop_listener(example_http).
+
+start_listeners() ->
+  {ok, Port} = application:get_env(example, http_port),
+  {ok, ListenerCount} = application:get_env(example, http_listener_count),
+  DescriptionTrail =
+    trails:trail(<<"/description">>
+                , example_description_handler
+                , []
+                , #{get => #{desc => "Retrives trails's server description"}}),
+  MessageTrail =
+  trails:trail(<<"/message/[:echo]">>
+              , example_echo_handler
+              , []
+              ,  #{ get => #{desc => "Gets echo var in the server"}
```

Comment:
it would be nice to add more fields to the metadata, to exemplify that as well. For instance:
```#{desc => "Gets echo var from the server", 'content-type' => "text/plain"}```